lib/Conversion/TorchToLinalg/TensorConstructors.cpp

```diff
@@ -79,6 +81,227 @@ class ConvertAtenConstantPadNdOp
+          SmallVector<int64_t> highPadding(4, 0);
+          highPadding[2] = padInts[3];
+          vLeftSlice = torch_to_linalg::getPaddedTensor(op, rewriter, vLeftSlice, lowPadding, highPadding, bottomLeftValue);
+        }
+        for (auto i=0; i<padInts[0]; ++i) {
+          tensorsLeft.push_back(vLeftSlice);
+        }
+        Value leftPadTile =
+            rewriter.create<tensor::ConcatOp>(loc, 3, tensorsLeft);
+        tensorsRes.push_back(leftPadTile);
+      }
+      if (hasTopPadding) {
+        Value topLeftValue = rewriter.create<tensor::ExtractOp>(
+            loc, input, ValueRange{zero, zero, zero, zero});
+        Value topHcenterSlice = rewriter.create<tensor::ExtractSliceOp>(
+            loc, input, extractOffsetsLT, extractShapeTB, allOneStrides);
+        for (auto i = 0; i < padInts[2]; ++i) {
+          tensorsCenter.push_back(topHcenterSlice);
+        }
+      }
+      tensorsCenter.push_back(input);
+      if (hasBottomPadding) {
+        Value bottomHcenterSlice = rewriter.create<tensor::ExtractSliceOp>(
+            loc, input, extractOffsetsBottom, extractShapeTB, allOneStrides);
+        for (auto i = 0; i < padInts[3]; ++i) {
+          tensorsCenter.push_back(bottomHcenterSlice);
+        }
+      }
+      centerTile = rewriter.create<tensor::ConcatOp>(loc, 2, tensorsCenter);
+      tensorsRes.push_back(centerTile);
+
+      if (hasRightPadding) {
+        Value vCenterRightSlice = rewriter.create<tensor::ExtractSliceOp>(
+            loc, input, extractOffsetsRight, extractShapeLR, allOneStrides);
+        Value vRightSlice = vCenterRightSlice;
+        if (hasTopPadding) {
+          Value topRightValue = rewriter.create<tensor::ExtractOp> (loc, input, ValueRange{zero, zero, zero, hDimSizeMinusOne});
+
+          //pad vCenterRightSlice on the top
+          SmallVector<int64_t> lowPadding(4, 0);
+          SmallVector<int64_t> highPadding(4, 0);
+          lowPadding[2] = padInts[2];
+          vRightSlice = torch_to_linalg::getPaddedTensor(op, rewriter, vRightSlice, lowPadding, highPadding, topRightValue);
+        }
+        if (hasBottomPadding) {
+          Value bottomRightValue = rewriter.create<tensor::ExtractOp> (loc, input, ValueRange{zero, zero, vDimSizeMinusOne, hDimSizeMinusOne});
+
+          //pad vCenterRightSlice or vRightTopPaddedSlice at the bottom
+          SmallVector<int64_t> lowPadding(4, 0);
+          SmallVector<int64_t> highPadding(4, 0);
+          highPadding[2] = padInts[3];
+          vRightSlice = torch_to_linalg::getPaddedTensor(op, rewriter, vRightSlice, lowPadding, highPadding, bottomRightValue);
+        }
+        for (auto i=0; i<padInts[1]; ++i) {
+          tensorsRight.push_back(vRightSlice);
+        }
+        Value rightPadTile = rewriter.create<tensor::ConcatOp>(loc, 3, tensorsRight);
+        tensorsRes.push_back(rightPadTile);
+      }     
+      Value resTensor = rewriter.create<tensor::ConcatOp>(loc, 3, tensorsRes);
```

Comment:
I left a comment on the PR for this commit, but unfortunately rebase/merge mix ups lost the comment: https://github.com/llvm/torch-mlir/commit/0860c41ee2a0bdec41f544f19eba170cf646c3ce

But the `tensor.pad` op was designed to be able to support this kind of dynamic padding: https://mlir.llvm.org/docs/Dialects/TensorOps/#tensorpad-tensorpadop
I'll say the same thing here as I did there, this is fine for now because I'm not sure how well the rest of the stack supports such pad ops, but it's likely we'll want to rework this lowering at some later point to use pad directly instead of this sequence of extracts/concats. It includes a body to allow dynamically determining the padding value to use based on the current indices.